learn/data_science/numpy.md

```diff
@@ -0,0 +1,255 @@
+  
+## What is `numpy` in Python  
+
+Numpy, standing for Numerical Python, is an integral part of the scientific computing environment in Python. It is a library that provides support for arrays, along with a rich collection of mathematical functions to perform various operations on these arrays. The [numpy](https://pypi.org/project/numpy/) library in Python is widely used in data analysis, machine learning, and engineering for its efficiency and functionality. It is designed to handle large multi-dimensional arrays and matrices, making it an indispensable tool for developers and researchers working in data-intensive domains. 
+
+### Importance of Numpy
+
+The `numpy` library in Python offers several features that make it essential for numerical computations:
+
+1. **High-performance multidimensional array object**: Numpy arrays are more compact, faster, and more convenient for mathematical operations than traditional Python lists.
+2. **Broad collection of mathematical functions**: These functions allow for efficient operations on arrays without the need for explicit loops.
+3. **Array broadcasting capabilities**: Numpy can process arrays of different shapes during arithmetic operations, making the code cleaner and faster.
+4. **Tools for integrating C/C++ and Fortran code**: It's easy to incorporate compiled code directly into Python applications for even more speed.
+
+> Note: Numpy's powerful data structures and computational tools are the foundation upon which libraries like Pandas, Matplotlib, SciPy, and scikit-learn are built.
+
+### Installing Numpy
+
+To make use of the [numpy](https://pypi.org/project/numpy/) library in Python, it first needs to be installed. This can typically be done using a Python package manager such as pip. The installation command is as follows:
+
+```python
+pip install numpy
+```
+
+## How to Start Using `numpy`  
+
+### Installing NumPy
+
+To begin using NumPy in your Python projects, the first step is **installing numpy**. NumPy can be installed using various package managers, but the most common and straightforward method is through pip, Python's package installer. Here's a step-by-step guide to **how to install [numpy](https://pypi.org/project/numpy/) in python**:
```

Comment:
Don't add a link to the same external page twice. We have the link to the https://pypi.org/project/numpy/ above.